runtime/array.c

```diff
@@ -370,20 +434,41 @@ static value caml_array_gather(intnat num_arrays,
     CAMLassert(pos == size);
   }
 #endif
-  else if (size > Max_wosize) {
-    /* Array of values, too big. */
-    caml_invalid_argument("Array.concat");
-  }
-  else {
+  else if (size <= Max_young_wosize) {
     /* Array of values, small enough to fit in young generation.
        We can use memcpy directly. */
     res = caml_alloc_small(size, 0);
     for (i = 0, pos = 0; i < num_arrays; i++) {
-      caml_blit_fields(arrays[i], offsets[i], res, pos, lengths[i]);
+      /* [res] is freshly allocated, and no other domain has a reference to it.
+         Hence, a plain [memcpy] is sufficient. */
+      memcpy(&Field(res, pos),
+             &Field(arrays[i], offsets[i]),
+             lengths[i] * sizeof(value));
```

Comment:
`memcpy` -> `wo_memcpy` ?

Reply:
`memcpy` is ok here since `res` is freshly allocated and no other domain yet has a reference to it. See the comment above.

Reply:
Right, sorry for having missed this point.